includes/data-events/listeners.php

```diff
@@ -301,3 +302,123 @@ function( $subscription, $status_to, $status_from ) {
 		];
 	}
 );
+
+/**
+ * When a non-donation subscription is activated.
+ * The subscription will be added to the user's list of active subscriptions.
+ */
+Data_Events::register_listener(
+	'woocommerce_subscription_status_active',
+	'product_subscription_active',
+	function( $subscription ) {
+		// We only want to fire this for non-donation products.
```

Comment:
If so, shouldn't the function return early if the subscription is for a donation product? I see it's handled later in `includes/reader-activation/class-reader-data.php`, but the comment gave me a pause.

Reply:
Good call, 6b26f574 returns early without firing the data event if there are no non-subscription products associated with the subscription.